packages/devtools_app/lib/src/framework/observer/memory_observer.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2025 The Flutter Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file or at https://developers.google.com/open-source/licenses/bsd.
+
+import 'dart:async';
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:devtools_app_shared/utils.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../devtools_app.dart';
+import '../../shared/analytics/analytics.dart' as ga;
+import '../../shared/analytics/constants.dart' as gac;
+import '../../shared/primitives/byte_utils.dart';
+import '_memory_desktop.dart' if (dart.library.js_interop) '_memory_web.dart';
+
+/// Observes the memory usage of the DevTools app (web only) and shows a memory
+/// pressure warning to users when DevTools is nearing the memory limit.
+///
+/// The warning has an option for users to automatically reduce memory. If the
+/// user selects this option, we will make a best effort attempt to clear data
+/// from features that have not been recently used, or stale data in general.
+class MemoryObserver extends DisposableController {
+  MemoryObserver({
+    @visibleForTesting this.debugMeasureUsageInBytes,
+    @visibleForTesting this.debugPollingDuration,
```

Comment:
nit:

```suggestion
    @visibleForTesting this.debugPollingDuration = _pollForMemoryDuration,
```

And then remove the `??` below.

Could just inline the value as well in the constructor and remove the static.